racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -7,20 +7,48 @@ SPDX-License-Identifier: GPL-3.0-or-later -->
 <script setup lang="ts">
 import { useHttp } from '@/plugins/http'
 import { useRacksDBAPI } from '@/composables/RacksDBAPI'
-import { ref, onMounted } from 'vue'
-import SearchBar from '@/components/SearchBar.vue'
+import { ref, onMounted, computed } from 'vue'
 import type { Ref } from 'vue'
-import type { Datacenter, Infrastructure, Rack } from '@/composables/RacksDBAPI'
+import type { Datacenter, Infrastructure } from '@/composables/RacksDBAPI'
+import { Switch, SwitchGroup, SwitchLabel } from '@headlessui/vue'
+import { BarsArrowDownIcon, BarsArrowUpIcon } from '@heroicons/vue/24/outline'
+import { Dialog, DialogPanel } from '@headlessui/vue'
 
 const http = useHttp()
 const racksDBAPI = useRacksDBAPI(http)
 const datacenters: Ref<Array<Datacenter>> = ref([])
 const infrastructures: Ref<Array<Infrastructure>> = ref([])
 const datacenterDetails: Ref<Datacenter | undefined> = ref()
-const racks: Ref<Array<Rack>> = ref([])
-const rackDetails: Ref<Array<Rack>> = ref([])
+const rackDetails: Ref<Array<any>> = ref([])
```

Comment:
It would be better to specify a real object type here instead of `any`. All the racks stored in this array contains the same keys with the same values types.

Reply:
I changed the type `any` with the type `Rack`.